test/unit/decorators/attribute.test.ts

```diff
@@ -206,3 +206,204 @@ describe(`@Attribute legacy decorator`, () => {
     ]);
   });
 });
+
+describe('@AllowNull legacy decorator', () => {
+  it('sets allowNull to true', () => {
+    class User extends Model<InferAttributes<User>> {
+      @Attribute(DataTypes.STRING)
+      @AllowNull
+      declare name: string;
+    }
+
+    sequelize.addModels([User]);
+
+    expect(User.getAttributes().name.allowNull).to.equal(true);
+  });
+
+  it('accepts a boolean', () => {
+    class User extends Model<InferAttributes<User>> {
+      @Attribute(DataTypes.STRING)
+      @AllowNull(false)
+      declare name: string;
+    }
+
+    sequelize.addModels([User]);
+
+    expect(User.getAttributes().name.allowNull).to.equal(false);
+  });
+});
+
+describe('@NotNull legacy decorator', () => {
+  it('sets allowNull to false', () => {
+    class User extends Model<InferAttributes<User>> {
+      @Attribute(DataTypes.STRING)
+      @NotNull
+      declare name: string;
+    }
+
+    sequelize.addModels([User]);
+
+    expect(User.getAttributes().name.allowNull).to.equal(false);
+  });
+
+  it('accepts a boolean', () => {
+    class User extends Model<InferAttributes<User>> {
+      @Attribute(DataTypes.STRING)
+      @NotNull(false)
```

Comment:
nit: May be worth only accepting booleans on `AllowNull` - IMO `NotNull` explicitly means "this cannot be null", whereas allow null means "Do we allow this to be null? Default to yes"

Reply:
I don't expect it to be used but I did it for symmetry with the other boolean decorators, such as `IsFloat`, `IsDecimal` etc

We could make a `createUnparameterizedAttributeValidationDecorator` and `createUnparameterizedAttributeOptionsDecorator`, but I figured we already had too many decorator-making functions

I think we just won't document it :)